app/src/main/java/com/mapbox/maps/testapp/examples/java/DSLStylingJavaActivity.java

```diff
@@ -155,7 +155,16 @@ private StyleContract.StyleExtension createStyle() {
         StyleExtensionImpl.Builder builder = new StyleExtensionImpl.Builder(Style.TRAFFIC_DAY);
 
         // Add a image source
-        builder.addSource(new ImageSource.Builder(IMAGE_URL).coordinates(POINT_LIST).build());
+        builder.addSource(
+                new ImageSource.Builder(IMAGE_SOURCE_ID)
+                        .coordinates(POINT_LIST)
+                        .url(IMAGE_URL)
+                        .build()
+        );
+        // Add a rasterLayer show the image
```

Comment:
Is the order of adding layers to the builder crucial here? Not sure I understand the fix

Reply:
the fix is that previously we were using `IMAGE_URL` as an id and not setting url at all 😄

Reply:
I see, thanks